eclipse-scout-core/src/desktop/hybrid/HybridManagerAdapter.ts

```diff
@@ -39,12 +40,111 @@ export class HybridManagerAdapter extends ModelAdapter {
   }
 
   protected _onWidgetHybridAction(event: HybridActionEvent) {
-    this._send('hybridAction', event.data);
+    this._send('hybridAction', {
+      actionType: event.data.actionType, // add as first property (devtools sometimes show properties in that order)
+      id: event.data.id,
+      data: event.data.data,
+      contextElements: this._contextElementsToJson(event.data.contextElements) || undefined
+    });
+  }
+
+  protected _jsonToContextElements(jsonContextElements: object): HybridActionContextElements {
+    if (!jsonContextElements) {
+      return null;
+    }
+    let contextElements = scout.create(HybridActionContextElements);
+    Object.keys(jsonContextElements).forEach(key => {
+      let list = this._jsonToContextElementList(jsonContextElements[key]);
+      contextElements.withElements(key, list);
+    });
+    return contextElements;
+  }
+
+  protected _jsonToContextElementList(jsonContextElements: JsonHybridActionContextElement[]): HybridActionContextElement[] {
+    if (!jsonContextElements) {
+      return null;
+    }
+    let list: HybridActionContextElement[] = [];
+    jsonContextElements.forEach(jsonContextElement => {
+      let contextElement = this._jsonToContextElement(jsonContextElement);
+      if (contextElement) {
+        list.push(contextElement);
+      }
+    });
+    return list;
+  }
+
+  protected _jsonToContextElement(jsonContextElement: JsonHybridActionContextElement): HybridActionContextElement {
+    if (!jsonContextElement) {
+      return null;
+    }
+    let adapter = this.session.getModelAdapter(jsonContextElement.widget);
+    if (!(adapter instanceof ModelAdapter)) {
+      throw new Error(`No adapter found for '${jsonContextElement.widget}'`);
+    }
+    let element = null;
+    if (jsonContextElement.element) {
+      element = HybridActionContextElementConverter.convertFromJson(adapter, jsonContextElement.element);
+      if (!element) {
+        throw new Error('Unable to convert JSON to context element');
+      }
+    }
+    return scout.create(HybridActionContextElement, {
+      widget: adapter.widget,
+      element: element
+    });
+  }
+
+  protected _contextElementsToJson(contextElements: HybridActionContextElements): object {
+    if (!contextElements || contextElements.isEmpty()) {
+      return null;
+    }
+    let json = {};
+    contextElements.map.forEach((list, key) => {
```

Comment:
order of arguments seems wrong at first, maybe for...of is easier to read
```suggestion
    for (const [key, list] of contextElements.map) {
```

Reply:
Yes, I was confused, too. The API designers clearly didn't think that through... 😢 

Your proposal is better, I changed it.